vllm/v1/sample/sampler.py

```diff
@@ -82,9 +80,17 @@ def sample(
     ) -> torch.Tensor:
         assert not (sampling_metadata.all_greedy
                     and sampling_metadata.all_random)
-        if sampling_metadata.all_greedy:
-            return self.greedy_sample(logits)
+        if sampling_metadata.all_random:
+            greedy_sampled = None
+        else:
+            greedy_sampled = self.greedy_sample(logits)
+            if sampling_metadata.all_greedy:
+                return greedy_sampled
 
+        # Apply temperature.
+        logits = self.apply_temperature(logits, sampling_metadata.temperature)
```

Comment:
Maybe worthwhile to add a comment here saying that this helps determinism?

Reply:
Oh maybe not. It's not related to the non-deterministic behavior. I misunderstood.